cmd/scollector/collectors/disk_linux.go

```diff
@@ -156,6 +163,56 @@ func c_iostat_linux() (opentsdb.MultiDataPoint, error) {
 	return md, err
 }
 
+// Check mdadm raid arrays on linux
+// linux.disk.mdadm.failed value:
+//   0 => active
+//   1 => something else, error
+func c_mdadm_linux() (opentsdb.MultiDataPoint, error) {
+	mdstat, err := os.Open("/proc/mdstat")
+	defer mdstat.Close()
+	if err != nil {
+		// there is no mdadm: not really an error
+		return opentsdb.MultiDataPoint{}, nil
+	}
+	return mdadmLinux(mdstat)
+}
+
+// mock our file for testing purpose
+func mdadmLinux(mdStat io.Reader) (opentsdb.MultiDataPoint, error) {
+	var md opentsdb.MultiDataPoint
+	var descr = "0: raid is active, 1: bad state (not active)"
+	scanner := bufio.NewScanner(mdStat)
+	for scanner.Scan() {
+		l := scanner.Text()
+		// md125 : active raid1 sda2[0] sdb2[1]
+		// s:0   1 2      3     disks...
+		if utf8.RuneCountInString(l) <= 2 || l[0:2] != "md" { // l =~ /^md/
+			continue
+		}
+		fields := strings.SplitN(l[2:], " ", 4)
+		if len(fields) <= 3 {
+			continue
+		}
+		// check that we really have an md device
+		if !IsDigit(fields[0]) {
+			continue
+		}
+		volumeName := "md" + fields[0]
+		raidFailed := 0 // 0: ok, 1: ko
+		if fields[2] != "active" {
+			raidFailed = 1
+		}
+
+		metric := "linux.disk.mdadm.failed"
+		tags := opentsdb.TagSet{"volume": volumeName}
+
+		// Now let's write some datapoints
+		// FIXME: do we have to write linux.disk. + os.disk.?
```

Comment:
For now I think it is fine to keep it in linux.disk, if we ever add software raid monitoring for another OS that would probably be the time to try to normalize it as an additional metric into the os.* namespace